lightning/mydump/router.go

```diff
@@ -143,14 +144,16 @@ func NewFileRouter(cfg []*config.FileRouteRule) (FileRouter, error) {
 // set value to target field in `RouteResult`
 type RegexRouter struct {
 	pattern    *regexp.Regexp
-	extractors []regexExpander
+	extractors []patExpander
 }
 
 func (r *RegexRouter) Route(path string) *RouteResult {
-	indexes := r.pattern.FindStringSubmatchIndex(path)
-	if len(indexes) == 0 {
+	// the regexp pattern maybe not contains any sub matches
+	if !r.pattern.MatchString(path) {
 		return nil
 	}
+
+	indexes := r.pattern.FindStringSubmatchIndex(path)
```

Comment:
???

even if `r.pattern` has no sub matches, if there is match `FindStringSubmatchIndex` will always return a non-empty array, since `$0` refers to the whole match.